src/ucp/core/ucp_mm.c

```diff
@@ -731,8 +732,9 @@ ucp_memh_reg_align(ucp_context_h context, void **address_p, size_t *length_p,
     end   = ucs_align_up_pow2_ptr(UCS_PTR_BYTE_OFFSET(*address_p, *length_p),
                                   reg_align);
 
-    *address_p = start;
-    *length_p  = UCS_PTR_BYTE_DIFF(start, end);
+    *address_p   = start;
+    *length_p    = UCS_PTR_BYTE_DIFF(start, end);
```

Comment:
do we still need to modify address/len? i'd expect rcache to do it now

Reply:
currently, it does it during region creation and lookup may return a region without proper alignment

Reply:
after this PR i'd expect to get address/len of the region from rcache, also we may have a merge..

Reply:
what if a region for a not-yet-aligned address/len already exists in rcache? it might have been created with a smaller alignment e.g. without gdrcopy in md_map and now we are adding gdrcopy. in this case lookup will return it as is and gdrcopy reg_mem will receive it unaligned

Reply:
maybe rcache lookup should fallback to slowpath if region->alignment is less than alignment param